### DISCUSSION TITLE
Unable to Compile Maven Project with Guava using j2cl-maven-plugin

### DISCUSSION THREAD

**Discussion by loveslikey** [Vertispan/j2clmavenplugin#269] [Q&A]
I am currently working on a project where I'm attempting to use the j2cl-maven-plugin to compile Java code for use in a browser environment. I've noticed that in the official demo, bazel-helloworld, libraries like Guava are being utilized successfully. However, when I try to import these same libraries (e.g., Guava ) into my Maven project using the j2cl-maven-plugin, the compilation fails.
I've searched for documentation or examples that might explain how to properly include these libraries in a Maven project with j2cl-maven-plugin, but haven't found anything relevant.

Could you please provide guidance on how to successfully use libraries like Guava with j2cl-maven-plugin in a Maven project? Is there a special configuration or additional steps that I need to follow?

Thank you for your assistance!
```
Error info:
ImmutableList cannot be resolved to a type [ERROR] com.major.j2cl:acss:1.0-SNAPSHOT/transpiled_js: Error:E:\project\idea_speace\j2clDemo\target\gwt3BuildCache\...\stripped_sources\App.java:39: ImmutableList cannot be resolved
```
maven dependency
` <dependency>\n <groupId>com.google.guava</groupId> <artifactId>guava-gwt</artifactId> <version>32.1.1-jre</version> </dependency>`

**Comment by niloc132** [comment]
This is new ground to cover - to my knowledge no one has investigated using guava in j2cl-maven-plugin yet. I believe that the bazel sample project that you referenced has some custom wiring for guava, it might be necessary to look into what that does exactly and seek to replicate it for this plugin.

Can you share the entire log, and confirm that the jar you have listed has the sources for ImmutableList.java (and that you do not need, for example, to have the sources jar added too)?

**Reply by loveslikey** [reply]
Thank you for your reply. I actually don't know how to configure Maven to support Guava properly for j2cl. I simply added the dependency to my Maven project:

```xml
<dependency>
    <groupId>com.google.guava</groupId>
    <artifactId>guava-gwt</artifactId>
    <version>30.1-jre</version>
</dependency>
```

Then in my Java code, I tried to use it like this:

```java
List<String> list = ImmutableList.of("a");
```

I'm not sure how to properly include the ImmutableList.java source code or if there are additional configuration steps needed for j2cl-maven-plugin to work with Guava. Could you provide any guidance on the correct way to set this up?

**Reply by loveslikey** [reply]
I want to achieve the same effect as in the j2cl samples: `https://github.com/google/j2cl/blob/master/samples/guava/src/main/java/com/google/j2cl/samples/guava/HelloGuava.java`
[link](https://github.com/google/j2cl/blob/master/samples/guava/src/main/java/com/google/j2cl/samples/guava/HelloGuava.java)
How can I implement this in my project? Specifically, how can I use Guava classes in my j2cl project? What steps should I take to make it work?

**Reply by niloc132** [reply]
Again, please share the full log, so that we might be able to ask other specific questions. Also consider sharing a full example project (on github, etc) so others can reproduce your work exactly?
